sstables/sstable_directory.cc

```diff
@@ -562,7 +562,7 @@ future<> sstable_directory::cleanup_column_family_temp_sst_dirs() {
         // so that the supplied callback will not block scan_dir() from
         // reading the next entry in the directory.
         fs::path dirpath = sstdir / de.name;
-        if (sstables::sstable::is_temp_dir(dirpath)) {
+        if (dirpath.extension().string() == tempdir_extension) {
             sstlog.info("Found temporary sstable directory: {}, removing", dirpath);
```

Comment:
this logic is completely tied to the storage type, to take advantage of XFS AGs. shouldn't we move it to filesystem_components_lister? possibly introducing a garbage_collect() to components_lister interface.

Reply:
perhaps components_lister needs to be more than just a lister, becoming a components_manager.

Reply:
> this logic is completely tied to the storage type, to take advantage of XFS AGs. shouldn't we move it to filesystem_components_lister?

Absolutely. That's what will happen in #13024 -- this part will be moved to filesystem lister and system-keyspace lister will get its own garbage collecting code